### PR TITLE
RPC Proxy

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-####

## What changed (plus any additional context for devs)
Switches all RPC endpoints to use [RPC proxy](https://github.com/rainbow-me/arc/pull/258). Requires `rpc_proxy_enabled` firebase flag to be true. If false, falls back to preexisting RPC endpoints.

Requires:
https://github.com/rainbow-me/rainbow-scripts/pull/136
https://github.com/rainbow-me/rainbow-env/pull/51

## What to test
1. Make sure I didn't forget to "convert" any RPC endpoints to use proxy.
2. Make sure proxy works (tx don't blow up)

### CODE REVIEW

File: src/env.ts [hunk 1]
```diff
@@ -1,5 +1,12 @@
 import ReactNative from 'react-native';
-import { ENABLE_DEV_MODE, IS_TESTING } from 'react-native-dotenv';
+import {
+  ENABLE_DEV_MODE,
+  IS_TESTING,
+  RPC_PROXY_BASE_URL_PROD,
+  RPC_PROXY_BASE_URL_DEV,
+  RPC_PROXY_API_KEY_PROD,
+  RPC_PROXY_API_KEY_DEV,
```
Comment (skylarbarrera) [inline src/env.ts:8]: gotta add these to globals.d.ts

File: src/model/config.ts [hunk 2]
```diff
@@ -85,19 +62,10 @@ const DEFAULT_CONFIG: RainbowConfig = {
...
   }),
-  ethereum_goerli_rpc: __DEV__ ? ETHEREUM_GOERLI_RPC_DEV : ETHEREUM_GOERLI_RPC,
-  ethereum_mainnet_rpc: __DEV__
-    ? ETHEREUM_MAINNET_RPC_DEV
-    : ETHEREUM_MAINNET_RPC,
   f2c_enabled: true,
   flashbots_enabled: true,
   op_nft_network: 'op-mainnet',
   op_rewards_enabled: false,
-  optimism_mainnet_rpc: OPTIMISM_MAINNET_RPC,
```
Comment (skylarbarrera) [inline src/model/config.ts:96]: if were gonna have this remotely configured it would make sense to keep these for a fallback i think

**Review by skylarbarrera — APPROVED** [review]
smol tings but lgtm

**Review by dereknelson — APPROVED** [review]
rpc stands for rainpow cool proxy

**Pull request merged**